pkg/controller/admissionchecks/multikueue/multikueuecluster.go

```diff
@@ -525,10 +529,69 @@ func (c *clustersReconciler) setupWithManager(mgr ctrl.Manager) error {
 		},
 	}
 
+	fsWatcherHndl := handler.Funcs{
+		GenericFunc: func(_ context.Context, e event.GenericEvent, q workqueue.RateLimitingInterface) {
+			// batch the events
+			q.AddAfter(reconcile.Request{NamespacedName: types.NamespacedName{
+				Name: e.Object.GetName(),
+			}}, 100*time.Millisecond)
+		},
+	}
+
+	filterLog := mgr.GetLogger().WithName("MultiKueueCluster filter")
+	filter := predicate.Funcs{
+		CreateFunc: func(ce event.CreateEvent) bool {
+			if cluster, isCluster := ce.Object.(*kueuealpha.MultiKueueCluster); isCluster {
+				if cluster.Spec.KubeConfig.LocationType == kueuealpha.PathLocationType {
+					err := c.fsWatcher.AddOrUpdate(cluster.Name, cluster.Spec.KubeConfig.Location)
+					if err != nil {
+						filterLog.V(2).Error(err, "AddOrUpdate FS watch", "cluster", klog.KObj(cluster))
```

Comment:
```suggestion
						filterLog.Error(err, "AddOrUpdate FS watch", "cluster", klog.KObj(cluster))
```
Also cleanup other places in the file